torchtune/datasets/multimodal/_llava_instruct.py

```diff
@@ -89,22 +94,34 @@ def __call__(self, sample: Mapping[str, Any]) -> Mapping[str, Any]:
                     role="system", content=self.new_system_prompt, masked=True, eot=True
                 )
             )
+
+        # Add in image stuffs / load from file
         for message in sample[self._column_map["conversations"]]:
             role = role_map[message["from"]]
+            content = message["value"]
             if role == "system" and self.new_system_prompt is not None:
                 continue
-            content = split_text_by_image_tag(message["value"], "<image>")
+            if role == "user":
+                pil_image = load_image(
+                    self.images_dir + sample[self._column_map["image"]]
+                )
+                content = format_content_with_images(
+                    content,
+                    image_tag="<image>",
+                    images=[pil_image],
+                )
             masked = (role != "assistant") and (not self.train_on_input)
             messages.append(Message(role=role, content=content, masked=masked))
 
-        return {"messages": messages, "images": sample[self._column_map["image"]]}
+        return {"messages": messages}
 
 
 # TODO: point to Flamingo model transform as an example
 def llava_instruct_dataset(
     model_transform: Transform,
     *,
     source: str = "liuhaotian/LLaVA-Instruct-150K",
+    images_dir: str = "coco/",
```

Comment:
I would love some input here. Right now this is a string b/c I'm anticipating usage from a config. However, the proper typing would be something like Path (Pathlib) b/c it would be much more tolerant to errant forward-slashes and plays better with combination of sub paths. 

But then, I'll have to do some conversion on the backside to do that. 

Thoughts? @RdoubleA @ebsmothers

Reply:
If we convert to Path on the backside but still maintain a string parameter, that should handle any trailing/incorrect slashes right? If so I would do that